samples/bluetooth/nrf_dm/src/main.c

```diff
@@ -116,6 +132,25 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
+static uint32_t get_id_addr_random_share(void)
+{
+	bt_addr_le_t addrs[CONFIG_BT_ID_MAX];
+	size_t count = CONFIG_BT_ID_MAX;
+
+	bt_id_get(addrs, &count);
+
+	__ASSERT(count == 1, "The sample assumes a single ID addr");
+
+	return scanner_addr_to_random_share(&addrs[0].a);
+}
+
+static uint32_t scanner_addr_to_random_share(bt_addr_t *p_scanner_addr)
+{
+	return (p_scanner_addr->val[0] | p_scanner_addr->val[1] << 8 |
```

Comment:
Just a question, I am not an expert here. Is it some kind of known hashing algorithm used here? I worried whether it gets a unique result for different BLE addresses

Reply:
It does not need to be unique. But there is indeed an issue with the code here. We should not use OR.

Reply:
actually. sorry. I think what we have is correct.
We don't need to guarantee uniqueness. We only need to make sure to create a uint32 from the scanner address.